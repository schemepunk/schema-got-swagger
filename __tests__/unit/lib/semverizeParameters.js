const SemverizeParameters = require('./../../../lib/SemverizeParameters');
const _ = require('lodash');
const { SchemaGotSwaggerError, SchemaGotSwaggerReThrownError } = require('./../../../lib/SchemaGotSwaggerError');
const semverishTest = require('./../../__helpers__/swaggerMainSemverish');
const Manifest = require('semverist/lib/supers/manifest');

let tmpMocks = [];

const semveristToPass = {
  semveristBehaviors: {
    inheritence: 'lazySemverist',
    lazySemverist: {
      preReleaseForwards: false,
      attributes: false,
    },
    default: false,
    defaultName: 'default',
    groups: false,
    mergeStrategy: 'merge',
    preReleasePattern: "[a-z]",
  },
  groups: {
    swagger: {
      members: [
        'swagger'
      ],
    },
  },
  prereleaseOrdering: {},
  directoryFileIgnorePattern: 'json$',
  converterType: 'default',
  semverishObjectLocation: '',
  composer: {
    composerType: 'default',
    destination: "",
    priority: 'default'
  }
}

describe('Constructor Tests', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
    jest.spyOn(Date, 'now').mockReturnValue(2000);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('Valid Construction with defaults.', () => {
    expect.assertions(9);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(semverizeParameters).toBeInstanceOf(SemverizeParameters);
    expect(semverizeParameters.semveristConfigtmp).toEqual({});
    expect(semverizeParameters.defaultSemveristConfig).toEqual('SgsSemverist');
    expect(semverizeParameters.dataDefaultsType).toEqual('');
    expect(semverizeParameters.tmpData).toEqual({test: 'example'});
    expect(semverizeParameters.desiredRealizations).toEqual([]);
    expect(semverizeParameters.targetName).toEqual('swagger');
    expect(semverizeParameters.semverishMolotov).toEqual({ overrides: {}, cocktailClasses: []});
    expect(semverizeParameters.validatorId).toEqual('UserInput');
  })
});

describe('Setters', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
    jest.spyOn(Date, 'now').mockReturnValue(2000);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('Valid semverist config is ok.', () => {
    expect.assertions(2);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(semverizeParameters.setSemveristConfig(semveristToPass)).toBeInstanceOf(SemverizeParameters)
    expect(semverizeParameters.getSemveristConfig()).toEqual(semveristToPass);
  });
  test('Invalid semverist config will throw', () => {
    expect.assertions(1);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(() => semverizeParameters.setSemveristConfig({nope: 'nope'})).toThrow(SchemaGotSwaggerError);
  });
  test('Set semverize Realizations with good realizations.', () => {
    expect.assertions(2);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(semverizeParameters.setSemverRealizations(['1.1.0'])).toBeInstanceOf(SemverizeParameters);
    expect(semverizeParameters.getSemverRealizations()).toEqual(['1.1.0']);
  });
  test('Invalid semverized realizations will throw', () => {
    expect.assertions(1);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(() => semverizeParameters.setSemverRealizations('nope')).toThrow(SchemaGotSwaggerError);
  });
  test('Validator with invalid validation name throws error', () => {
    expect.assertions(1)
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    expect(() => semverizeParameters.validator('nope', {test: 'example'})).toThrow(SchemaGotSwaggerError);
  })
});

describe('Composer related', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
    jest.spyOn(Date, 'now').mockReturnValue(2000);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('Create composer..', () => {
    expect.assertions(5);
    const semverizeParameters = new SemverizeParameters(
      {apiName: 'Swagger Example APi override'},
      'UserInput'
    );

    return SemverizeParameters.createComposer(semverishTest, { test: semveristToPass }, 'test')
    .then(value => {
      expect(value.getComposition()).toMatchSnapshot()
      semverizeParameters.parseComposerAttributes(value);
      semverizeParameters.targetName = 'swaggerSrc';
      semverizeParameters.validatorId = 'swaggerMainSrcValidator';
      semverizeParameters.dataDefaultsType = 'UserInput';
      expect(semverizeParameters.getSemverRealizations()).toEqual(
        [
          "1.0.0",
          "1.1.0",
          "1.1.1"
        ]
      );
      expect(semverizeParameters.realized).toMatchSnapshot();
      expect(semverizeParameters.composer).toBeInstanceOf(Manifest);
      expect(semverizeParameters.validateRealizedParameters()).toBeInstanceOf(SemverizeParameters);
    });

  });
});

describe('Prepare semverish', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
    jest.spyOn(Date, 'now').mockReturnValue(2000);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('Prepare semverish pass in semverish.', () => {
    expect.assertions(1);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );

    return semverizeParameters.prepareSemverish(semverishTest)
    .then(value => expect(value).toEqual(semverishTest));
  });

  test('Prepare semverish, non semverish', () => {
    expect.assertions(1);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    semverizeParameters.dataDefaultsType = 'SwaggerSrcTemplates';
    semverizeParameters.targetName = 'swagger';
    semverizeParameters.desiredRealizations = ['1.1.0', '2.0.0'];
    return semverizeParameters.prepareSemverish({test: 'example'})
    .then(value => expect(value).toMatchSnapshot());
  });

  test('Semverize To Realizations.', () => {
    expect.assertions(1);
    const semverizeParameters = new SemverizeParameters(
      {test: 'example'},
      'UserInput'
    );
    semverizeParameters.desiredRealizations = ['1.1.0', '2.0.0'];
    expect(semverizeParameters.semverizeToRealizations({test: 'example'})).toEqual(
      {"1": {"1": {"0": {"test": "example"}}}, "2": {"0": {"0": {"test": "example"}}}}
    );
  });
});

