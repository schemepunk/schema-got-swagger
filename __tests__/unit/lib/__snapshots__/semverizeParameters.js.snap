// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composer related Create composer.. 1`] = `
Object {
  "1": Object {
    "0": Object {
      "0": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
    },
    "1": Object {
      "0": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
      "1": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
    },
  },
}
`;

exports[`Composer related Create composer.. 2`] = `
Object {
  "1": Object {
    "0": Object {
      "0": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
    },
    "1": Object {
      "0": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
      "1": Object {
        "swaggerSrc": Object {
          "apiBasePath": "/api",
          "apiDescription": "An example Api for example corp.",
          "apiHost": "example.com",
          "apiName": "Swagger Example APi",
          "consumes": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "contact": Object {
            "email": "example@example.com",
            "name": "Dave Example",
            "url": "http://example.com",
          },
          "definitions": Object {
            "items": Object {
              "example": Object {
                "properties": Object {
                  "id": Object {
                    "format": "int64",
                    "type": "integer",
                  },
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
              "example2": Object {
                "properties": Object {
                  "name": Object {
                    "type": "string",
                  },
                },
                "type": "object",
              },
            },
            "process": [Function],
          },
          "externalDocs": Object {
            "description": "Further api docs",
            "url": "http://example.com/docs",
          },
          "license": Object {
            "name": "apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
          },
          "parameters": Object {
            "items": Object {
              "skipParam": Object {
                "description": "number of items to skip",
                "format": "int32",
                "in": "query",
                "name": "skip",
                "required": true,
                "type": "integer",
              },
            },
            "process": [Function],
          },
          "produces": Array [
            Object {
              "mime": Object {
                "comma": false,
                "type": "vnd.api+json",
              },
            },
          ],
          "responses": Object {
            "items": Object {
              "GeneralError": Object {
                "description": "General Error",
              },
              "IllegalInput": Object {
                "description": "Illegal input for operation.",
              },
              "notFound": Object {
                "description": "Entity not found.",
              },
            },
            "process": [Function],
          },
          "schemes": Object {
            "items": Array [
              "https",
              "http",
            ],
            "process": [Function],
          },
          "security": Object {
            "items": Object {
              "petstore_auth": Array [
                "write:pets",
                "read:pets",
              ],
            },
            "process": [Function],
          },
          "securityDefinitions": Object {
            "items": Object {
              "api_key": Object {
                "in": "header",
                "name": "api_key",
                "type": "apiKey",
              },
              "petstore_auth": Object {
                "authorizationUrl": "http://swagger.io/api/oauth/dialog",
                "flow": "implicit",
                "scopes": Object {
                  "read.pets": "read your pets",
                  "write.pets": "modify pets in your account",
                },
                "type": "oauth2",
              },
            },
            "process": [Function],
          },
          "tags": Object {
            "items": Array [
              Object {
                "description": "Pets operations",
                "externalDocs": Object {
                  "description": "pet docs",
                  "url": "http://example.com/pet",
                },
                "name": "pet",
              },
              Object {
                "description": "Vegetable",
                "name": "carrot",
              },
            ],
            "process": [Function],
          },
          "termsOfService": "",
        },
      },
    },
  },
}
`;

exports[`Prepare semverish Prepare semverish, non semverish 1`] = `
Object {
  "1": Object {
    "1": Object {
      "0": Object {
        "swagger": Object {
          "composer": Object {
            "composerType": "default",
            "destination": "",
            "priority": "default",
          },
          "converterType": "default",
          "directoryFileIgnorePattern": "json$",
          "groups": Object {
            "example": Object {
              "members": Array [
                "example",
              ],
            },
          },
          "prereleaseOrdering": Object {},
          "semverishObjectLocation": "",
          "semveristBehaviors": Object {
            "default": false,
            "defaultName": "default",
            "groups": false,
            "inheritence": "lazySemverist",
            "lazySemverist": Object {
              "attributes": false,
              "preReleaseForwards": false,
            },
            "mergeStrategy": "merge",
            "preReleasePattern": "[a-z]",
          },
          "test": "example",
        },
      },
    },
  },
  "2": Object {
    "0": Object {
      "0": Object {
        "swagger": Object {
          "composer": Object {
            "composerType": "default",
            "destination": "",
            "priority": "default",
          },
          "converterType": "default",
          "directoryFileIgnorePattern": "json$",
          "groups": Object {
            "example": Object {
              "members": Array [
                "example",
              ],
            },
          },
          "prereleaseOrdering": Object {},
          "semverishObjectLocation": "",
          "semveristBehaviors": Object {
            "default": false,
            "defaultName": "default",
            "groups": false,
            "inheritence": "lazySemverist",
            "lazySemverist": Object {
              "attributes": false,
              "preReleaseForwards": false,
            },
            "mergeStrategy": "merge",
            "preReleasePattern": "[a-z]",
          },
          "test": "example",
        },
      },
    },
  },
}
`;
